src/components/OverviewCard.jsx

```diff
@@ -11,9 +23,131 @@ import {
   retrieveResourceFileMetadata,
 } from '../utils';
 
+// axios settings
+axios.defaults.withCredentials = true
+
 const OverviewCard = ({
   date, category, settingsToggle, itemIndex, siteName, fileName, isResource, isHomepage
 }) => {
+  const dropdownRef = useRef(null)
+  const fileMoveDropdownRef = useRef(null)
+  const categoryInputRef = useRef(null)
+  const [allCategories, setAllCategories] = useState()
+  const [newCategory, setNewCategory] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+  const [canShowDropdown, setCanShowDropdown] = useState(false)
+  const [canShowFileMoveDropdown, setCanShowFileMoveDropdown] = useState(false)
+  const [canShowDeleteWarningModal, setCanShowDeleteWarningModal] = useState(false)
+  const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${category ? isResource ? `/resources/${category}` : `/collections/${category}` : ''}`
+
+  useEffect(() => {
+    const fetchData = async () => {
+      // Retrieve the list of all page/resource categories for use in the dropdown options.
+      if (isResource) {
+        const resourcesResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources`);
+        const { resources: allCategories } = resourcesResp.data;
+        setAllCategories(allCategories.map((category) => category.dirName))
+      } else {
+        const collectionsResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/collections`);
+        const { collections: collectionCategories } = collectionsResp.data;
+        setAllCategories(collectionCategories)
+      }
+    }
+    fetchData()
+  }, [])
+
+  useEffect(() => {
+    if (canShowFileMoveDropdown) fileMoveDropdownRef.current.focus()
+    if (canShowDropdown) dropdownRef.current.focus()
+  }, [canShowFileMoveDropdown, canShowDropdown])
+
+  const moveFile = async (chosenCategory) => {
+    try {
+      // Retrieve data from existing page/resource
+      const resp = await axios.get(`${baseApiUrl}/pages/${fileName}`);
+
+      const { content, sha } = resp.data;
+      const base64DecodedContent = Base64.decode(content);
+      const { frontMatter, mdBody } = frontMatterParser(base64DecodedContent);
+      const {
+        title, permalink, file_url: fileUrl, date, third_nav_title: thirdNavTitle,
+      } = frontMatter;
+
+      const { editableLink } = retrieveCollectionAndLinkFromPermalink(permalink)
+
+      const fileInfo = {
+        title,
+        permalink: editableLink,
+        fileUrl,
+        date,
+        mdBody,
+        sha,
+        category: chosenCategory ? chosenCategory : newCategory,
+        prevCategory: category,
+        baseApiUrl,
+        type: isResource ? 'resource' : 'page',
+        thirdNavTitle,
+        fileName,
+        isNewFile: false,
+        siteName,
+      }
+      await saveFileAndRetrieveUrl(fileInfo)
```

Comment:
There are problems with the `saveFileAndRetrieveUrl` function when you try to move a file from unlinked pages to a collection - the new file name generated is wrong, the group identifier is not generated. For example, if I moved a page titled `bah-gawd.md` to the `left-nav-one` collection, it is renamed `bah-bah-gawd.md`. PR #218 has been updated with new logic so that might help to resolve things

Reply:
Resolved after rebase, using the util function from PR #218

Reply:
This doesn't work when trying to move a page from workspace to a collection - the calculated `newBaseApiUrl` is wrong, it attempts to delete the original file from the collection it is being moved into

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/219/commits/a161fafdc75f087226ecdbdf1392f0d357c7be2c